library/lib/core/application.cpp

```diff
@@ -159,10 +161,60 @@ bool Application::mainLoop()
 
     // Input
     ControllerState controllerState = {};
+    RawTouchState rawTouch          = {};
 
     InputManager* inputManager = Application::platform->getInputManager();
+    inputManager->updateTouchState(&rawTouch);
     inputManager->updateControllerState(&controllerState);
 
+    static TouchState oldTouch;
+    TouchState touchState = InputManager::computeTouchState(rawTouch, oldTouch);
+    oldTouch              = touchState;
+
+    // Touch controller events
+    switch (touchState.phase)
+    {
+        case TouchPhase::START:
+            Logger::debug("Touched at X: " + std::to_string(touchState.position.x) + ", Y: " + std::to_string(touchState.position.y));
+            Application::setInputType(InputType::TOUCH);
+
+            // Search for first responder, which will be the root of recognition tree
+            if (Application::activitiesStack.size() > 0)
+                firstResponder = Application::activitiesStack[Application::activitiesStack.size() - 1]
+                                     ->getContentView()
+                                     ->hitTest(touchState.position);
+            break;
+        case TouchPhase::NONE:
+            firstResponder = nullptr;
+            break;
+        default:
+            break;
+    }
+
+    if (!firstResponder && (touchState.scroll.x != 0 || touchState.scroll.y != 0))
+    {
+        Logger::debug("Touched at X: " + std::to_string(touchState.position.x) + ", Y: " + std::to_string(touchState.position.y));
+        Application::setInputType(InputType::TOUCH);
+
+        // Search for first responder, which will be the root of recognition tree
+        if (Application::activitiesStack.size() > 0)
+            firstResponder = Application::activitiesStack[Application::activitiesStack.size() - 1]
+                                 ->getContentView()
+                                 ->hitTest(touchState.position);
+    }
+
+    if (firstResponder)
+    {
+        Sound sound = firstResponder->gestureRecognizerRequest(touchState, firstResponder);
+        float pitch = 1;
+        if (sound == SOUND_TOUCH)
```

Comment:
IIRC in HOS there are more sounds that have their pitch randomized, did you poke around to see for yourself?

Reply:
Could you hint me what else sounds have randomized pitch? It was the only sound I heard randomized.